torch/csrc/jit/codegen/cuda/mutator.cpp

```diff
@@ -149,21 +149,21 @@ void OptOutMutator::mutate(Expr* op) {
     if (!all_same) {
       break;
     }
-    all_same = all_same && mutated_outputs[i]->sameAs(op->output(i));
+    all_same = all_same && mutated_outputs[i] == op->output(i);
```

Comment:
Why is this changed?

Reply:
For example, if I have
```
T1 = sin(T0)
T2 = sin(T0)
T3 = sum(T2, keepdim=True) + T2
```
And I want to replace `T2` with `T1` in the definition `T3`, clearly what I want is pointer check, not `sameAs`, because `T2->sameAs(T1)`. In the past, `sameAs` worked as well, because for `Val`, it was just checking pointers. But after I fixed this TODO:
```C++
  // TODO: Make this more sophisticated. A value being the same as another value
  // should be evaluated based on the DAG that created it, and that DAGs leaf
  // nodes
```
`sameAs` stopped working.